### DISCUSSION TITLE
Embedding commit history on pages

### DISCUSSION THREAD

**Discussion by Bassoonian** [community/community#22234] [Other Feature Feedback, Questions, & Ideas]
<p>I host a page on GitHub pages and wanted to embed a changelog on the page that would be pulled from the repo commit history. I wasn’t sure if I could use the GitHub API through GitHub pages, so figured out a possible hacky way was to request the HTML of the commit page through $.get() and to go through it, parsing it and filtering out the information I need. This worked fine until I pushed it to GitHub - I get the following error:</p>
<p>Access to XMLHttpRequest at ‘<a href="https://github.com/Bassoonian/bassoonian.github.io/commits/master" rel="noopener nofollow ugc">https://github.com/Bassoonian/bassoonian.github.io/commits/master</a>’ from origin ‘<a href="https://bassoonian.github.io" rel="noopener nofollow ugc">https://bassoonian.github.io</a>’ has been blocked by CORS policy: No ‘Access-Control-Allow-Origin’ header is present on the requested resource.</p>
<p>Is there some other way to gain access to the commit history of a given repository, either through the API or through some other means, and to display it on GitHub Pages? Thank you in advance!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Bassoonian" data-post="1" data-topic="146050">
<div class="title">
<div class="quote-controls"></div>
 Bassoonian:</div>
<blockquote>
<p>I wasn’t sure if I could use the GitHub API through GitHub pages</p>
</blockquote>
</aside>
<p>You should be able to, as long as you do it in a script on the web site itself. However, there’s pretty strict <a href="https://docs.github.com/en/free-pro-team@latest/rest/overview/resources-in-the-rest-api#rate-limiting">rate limiting</a> for unauthenticated requests.</p>
<p>An alternative might be to use Actions to statically build your changelog page from the commit history. Use <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> with <code>fetch-depth: 0</code>, parse whatever you need from the repository and generate the page, and then commit &amp; push.</p>
